flow/JsonString.cpp

```diff
@@ -0,0 +1,303 @@
+	return *this;
+}
+
+JsonString& JsonString::append( const std::string& value ) {
+	return appendImpl(value, true);
+}
+JsonString& JsonString::append( const char* value ) {
+	return appendImpl(stringify(value), true);
+}
+JsonString& JsonString::append( double value ) {
+	return appendImpl(stringify(value), false);
+}
+JsonString& JsonString::append( long int value ) {
+	return appendImpl(stringify(value), false);
+}
+JsonString& JsonString::append( long unsigned int value ) {
+	return appendImpl(stringify(value), false);
+}
+JsonString& JsonString::append( long long int value ) {
+	return appendImpl(stringify(value), false);
+}
+JsonString& JsonString::append( long long unsigned int value ) {
+	return appendImpl(stringify(value), false);
+}
+JsonString& JsonString::append( int value ) {
+	return appendImpl(stringify(value), false);
+}
+JsonString& JsonString::append( unsigned value ) {
+	return appendImpl(stringify(value), false);
+}
+JsonString& JsonString::append( bool value ) {
+	return appendImpl(stringify(value), false);
+}
+JsonString& JsonString::append( const JsonString& value ) {
+	// Only do something, if not empty
+	if (!value.empty()) {
+		if (!_jsonText.empty())
+			_jsonText += ",\n  ";
+		_jsonText += value._jsonText;
+		_keyNames.insert(value._keyNames.begin(), value._keyNames.end());
+	}
+	return *this;
+}
+JsonString& JsonString::append( const JsonStringArray& values ) {
+	_jsonText += _jsonText.empty() ? "[ " : ",\n  [ ";
+	size_t counter = 0;
+	for (auto const& value : values) {
+		if (counter)
+			_jsonText += ",\n  ";
+		_jsonText += value.getJson();
+		counter ++;
+	}
+	_jsonText += " ]";
+	return *this;
+}
+
+JsonString& JsonString::clear() {
+	_keyNames.clear();
+	_jsonText.clear();
+	return *this;
+}
+
+bool	JsonString::empty() const {
+	return _jsonText.empty();
+}
+
+const std::string&	JsonString::getJsonText() const {
+	return _jsonText;
+}
+
+size_t	JsonString::getLength() const {
+	return _jsonText.length() + ((!empty() && _keyNames.empty()) ? 0 : 2);
+}
+size_t	JsonString::getNameTotal() const {
+	return _keyNames.size();
+}
+
+std::string	JsonString::getJson() const {
+	// If not empty with no names (only values), don't add brackets because prob in an array
+	return (!empty() && _keyNames.empty()) ? _jsonText : ("{ " + _jsonText + " }");
+}
+
+JsonString& JsonString::copy( const JsonString& jsonString ) {
+	_jsonText = jsonString._jsonText;
+	_keyNames = jsonString._keyNames;
+	return *this;
+}
+
+JsonString& JsonString::operator=( const JsonString& jsonString ) {
+	return copy(jsonString);
+}
+
+//TODO: Populate key names member
+void	JsonString::setJson(const std::string& jsonText) {
+	_keyNames.clear();
+	_jsonText = jsonText;
+}
+JsonString&	JsonString::swapJsonText(std::string& jsonText) {
+	_keyNames.clear();
+	_jsonText.swap(jsonText);
+	return *this;
+}
+
+JsonStringSetter& JsonString::operator[]( const std::string& name ) {
+	JsonStringSetter* stringSetter = new JsonStringSetter(*this, name);
```

Comment:
This is a memory leak, it's allocating a new object on the heap and then returning a reference to it.